### PR TITLE
Get libpalaso dependencies from Nuget instead of TC (BL-9332) 

### PR DESCRIPTION
**Pull request by sujeffreyl**
This change is [Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/4111)

### CODE REVIEW

**Review by andrew-polk — CHANGES_REQUESTED** [review]
Thanks for doing this.

So a follow-up change will be to update the getDependencies scripts after the TC config is modified, right?

Have you looked at the installer at all? 
Tested?
Looked at size change?
Evaluated if anything is going in which isn't needed? 

> Reviewed 8 of 8 files at r1.
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/4111#-:-MOYKHvx5z5M9F_rebiI:bn7bxqt)* status: all files reviewed, 3 unresolved discussions (waiting on @sujeffreyl)

___
*[src/BloomExe/BloomExe.csproj, line 122 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4111#-MOYJDr-3P_HtIyBD5IN:-MOYJDr02D5S-bmrRPRg:bp4qew1) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/edf05f3c0dbc597e7ac4ecc458eb17f36b52348b/src/BloomExe/BloomExe.csproj#L122)):*
> ```
> DialogAdapters.Gtk2
> ```
Is this because SIL.something pulls it in? It seems like libpalaso should be using DialogAdapters package.
Even if I'm wrong or it doesn't matter, is this really going to work, having two references with the same name?
___
*[src/BloomExe/BloomExe.csproj, line 306 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4111#-MOYJtcc5xJdhnuHEy4V:-MOYJtcc5xJdhnuHEy4W:b-tgisqy) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/edf05f3c0dbc597e7ac4ecc458eb17f36b52348b/src/BloomExe/BloomExe.csproj#L306)):*
> ```XML
>     <Reference Include="TagLibSharp, Version=2.2.0.0, Culture=neutral, PublicKeyToken=db62eba44689b5b0, processorArchitecture=MSIL">
>       <HintPath>..\..\packages\TagLibSharp.2.2.0\lib\net45\TagLibSharp.dll</HintPath>
>     </Reference>
> ```

With NAudio, I know we are ok going to NuGet because that was verified recently. 
Have you evaluated that we aren't losing anything going  to NuGet for TagLibSharp? For example, are we building our own fork currently?
___
*[src/BloomExe/Linux/packages.config, line 13 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4111#-MOYI6VO6mLgLnY4kCci:-MOYI6VO6mLgLnY4kCcj:btoaywi) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/edf05f3c0dbc597e7ac4ecc458eb17f36b52348b/src/BloomExe/Linux/packages.config#L13)):*
> ```
> TODO
> ```
Even if we don't answer this right now, let's move it down into the Linux-only section or its own section.
In other words, this is a good opportunity to get the section which is supposed to be in sync into sync.

Same below.

**Review by sujeffreyl — COMMENTED** [review]
getDependencies: I was halfway through that, forget to check it in. Updated w/Win and Linux versions

> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/4111#-:-MOYSgScAoUBcE9BKJUj:b6sad9n)* status: 6 of 10 files reviewed, 3 unresolved discussions (waiting on @andrew-polk and @sujeffreyl)

___
*[src/BloomExe/BloomExe.csproj, line 122 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4111#-MOYJDr-3P_HtIyBD5IN:-MOYLHTd7khTH8Ck4BvW:b-k5c8gz) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/edf05f3c0dbc597e7ac4ecc458eb17f36b52348b/src/BloomExe/BloomExe.csproj#L122)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

> ```
> DialogAdapters.Gtk2
> ```
Is this because SIL.something pulls it in? It seems like libpalaso should be using DialogAdapters package.
Even if I'm wrong or it doesn't matter, is this really going to work, having two references with the same name?


</blockquote></details>

Yes, it worked, I think because of the bindingRedirect.
Anyhow, turns out that while an older beta of SIL.Windows.Forms asked for DialogAdapters.Gtk2, the most recent one asks for DialogAdapters (i.e. GTK3), so that makes this less messy. Yay
___
*[src/BloomExe/Linux/packages.config, line 13 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4111#-MOYI6VO6mLgLnY4kCci:-MOYPtRiBEVvoho37STU:b-896fix) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/edf05f3c0dbc597e7ac4ecc458eb17f36b52348b/src/BloomExe/Linux/packages.config#L13)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

> ```
> TODO
> ```
Even if we don't answer this right now, let's move it down into the Linux-only section or its own section.
In other words, this is a good opportunity to get the section which is supposed to be in sync into sync.

Same below.
</blockquote></details>

Done.

**Review by sujeffreyl — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/4111)* status: 6 of 10 files reviewed, 3 unresolved discussions (waiting on @andrew-polk)

___
*[src/BloomExe/BloomExe.csproj, line 306 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4111#-MOYJtcc5xJdhnuHEy4V:-MOY_NfC-VVjOg2LFoiw:b4m5zl4) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/edf05f3c0dbc597e7ac4ecc458eb17f36b52348b/src/BloomExe/BloomExe.csproj#L306)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

With NAudio, I know we are ok going to NuGet because that was verified recently. 
Have you evaluated that we aren't losing anything going  to NuGet for TagLibSharp? For example, are we building our own fork currently?
</blockquote></details>

What i've found about TagLibSharp so far:
Bloom master gets taglib-sharp 2.1.2.0, last build April 14, 2015 (5.5 years ago!), claims to be built from https://github.com/sillsdev/taglib-sharp. I'm confused though because that repo has way more commits since 2015 than I see in Teamcity...

In the new one, we get the official TagLibSharp nuget package 2.2.0.0, which was last updated July 2019. I doubt we're missing anything, we could be adding lots of things though...

**Review by andrew-polk — CHANGES_REQUESTED** [review]
> Reviewed 4 of 4 files at r2.
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/4111)* status: all files reviewed, 1 unresolved discussion (waiting on @sujeffreyl)

___
*[src/BloomExe/BloomExe.csproj, line 306 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4111#-MOYJtcc5xJdhnuHEy4V:-MOYiP--4hqb0xaGYEMl:bpxcr8k) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/edf05f3c0dbc597e7ac4ecc458eb17f36b52348b/src/BloomExe/BloomExe.csproj#L306)):*
<details><summary><i>Previously, sujeffreyl wrote…</i></summary><blockquote>

What i've found about TagLibSharp so far:
Bloom master gets taglib-sharp 2.1.2.0, last build April 14, 2015 (5.5 years ago!), claims to be built from https://github.com/sillsdev/taglib-sharp. I'm confused though because that repo has way more commits since 2015 than I see in Teamcity...

In the new one, we get the official TagLibSharp nuget package 2.2.0.0, which was last updated July 2019. I doubt we're missing anything, we could be adding lots of things though...
</blockquote></details>

Ok, I'll take a look.
___
*[src/BloomTests/BloomTests.csproj, line 103 at r2](https://reviewable.io/reviews/bloombooks/bloomdesktop/4111#-MOYiLTc5wnmCR-r-9XA:-MOYiLTc5wnmCR-r-9XB:b-a8nkht) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/af6863a42e6f14b8822a58a8e740eb3a93997726/src/BloomTests/BloomTests.csproj#L103)):*
> ```
> <HintPath>..\..\packages\DialogAdapters.Gtk2.0.1.9\lib\net461\DialogAdapters.dll</HintPath>
> ```
This looks wrong.

**Review by andrew-polk — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/4111)* status: all files reviewed, 2 unresolved discussions (waiting on @sujeffreyl)

**Review by andrew-polk — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/4111)* status: all files reviewed, 2 unresolved discussions (waiting on @sujeffreyl)

___
*[src/BloomExe/BloomExe.csproj, line 306 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4111#-MOYJtcc5xJdhnuHEy4V:-MOYjneP7s2hIxIxx8yy:b7ii34c) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/edf05f3c0dbc597e7ac4ecc458eb17f36b52348b/src/BloomExe/BloomExe.csproj#L306)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

Ok, I'll take a look.
</blockquote></details>

I didn't realize this was coming in from libpalaso originally, too. I've asked on Slack.

**Review by andrew-polk — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/4111)* status: all files reviewed, 1 unresolved discussion (waiting on @sujeffreyl)

___
*[src/BloomExe/BloomExe.csproj, line 306 at r1](https://reviewable.io/reviews/bloombooks/bloomdesktop/4111#-MOYJtcc5xJdhnuHEy4V:-MOapZQNCWf6bYsMky1z:bvtg72v) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/edf05f3c0dbc597e7ac4ecc458eb17f36b52348b/src/BloomExe/BloomExe.csproj#L306)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

I didn't realize this was coming in from libpalaso originally, too. I've asked on Slack.
</blockquote></details>

Eberhard confirmed this is ok.

**Review by sujeffreyl — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/4111)* status: 8 of 10 files reviewed, 1 unresolved discussion (waiting on @andrew-polk)

___
*[src/BloomTests/BloomTests.csproj, line 103 at r2](https://reviewable.io/reviews/bloombooks/bloomdesktop/4111#-MOYiLTc5wnmCR-r-9XA:-MObAwZ83pFmynwSGIm3:b-896fix) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/af6863a42e6f14b8822a58a8e740eb3a93997726/src/BloomTests/BloomTests.csproj#L103)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

> ```
> <HintPath>..\..\packages\DialogAdapters.Gtk2.0.1.9\lib\net461\DialogAdapters.dll</HintPath>
> ```
This looks wrong.
</blockquote></details>

Done.

**Review by andrew-polk — COMMENTED** [review]
Changes look good. We just need to verify the installer is good.

> Reviewed 3 of 3 files at r3.
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/4111#-:-MObL5cL2p8Fe7f1PLDk:b-94nc02)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @sujeffreyl)

___
*[src/BloomTests/BloomTests.csproj, line 103 at r2](https://reviewable.io/reviews/bloombooks/bloomdesktop/4111#-MOYiLTc5wnmCR-r-9XA:-MObKyE65n5uXQH2NjR0:bz7llk4) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/af6863a42e6f14b8822a58a8e740eb3a93997726/src/BloomTests/BloomTests.csproj#L103)):*
<details><summary><i>Previously, sujeffreyl wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Somehow we lost the leading dot in the path, too

**Review by andrew-polk — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/4111)* status: all files reviewed, 1 unresolved discussion (waiting on @sujeffreyl)

**Review by andrew-polk — COMMENTED** [review]
> Reviewed 1 of 1 files at r4.
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/4111)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @sujeffreyl)

**Review by sujeffreyl — COMMENTED** [review]
I was able to build a local installer. It installed and I was able to run it, make some books, record audio, split audio, publish etc. The installer size is up 3% from roughly 93 MB to 96 MB.

> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/4111#-:-MObfb9gE4cZrIigiRx7:bj3cae2)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @sujeffreyl)

**Review by sujeffreyl — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/4111#-:-MObh5iN-bCw5AAI24VZ:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @sujeffreyl)
